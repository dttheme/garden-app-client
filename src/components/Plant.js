import React, { Component } from "react";
import { Link } from 'react-router-dom';
// var FontAwesome = require('react-fontawesome');
//
// <FontAwesome
//   className="fa fa-calendar-alt"
// />

const Plant = (props) => {
  return (
    <div className="">
      <h3
        className="f4 fw9 tracked"
        >
        {!props.isNotLinking ? <Link className="washed-green no-underline underline-hover hover-light-yellow" to={`/plant/${props.plant._id}`}>{props.plant.plantName}</Link> : props.plant.plantName} ({props.plant.numberPlanted})
      </h3>
      <p>
        {props.plant.plantDate}</p>
      <p>Location: {props.plant.plantLocation}</p>
      <p>You should water: {props.plant.waterFrequency}</p>
      {props.deleteHandler && <button onClick={()=>props.deleteHandler(props.plant._id)}>x</button>}
  </div>
);
}

Plant.defaultProps = {
  plantName: "",
  plantDate: "00/00/00",
  numberPlanted: null,
  plantLocation: "",
  waterFrequency: "",
  deleteHandler: null,
  isNotLinking: false
}

export default Plant;
