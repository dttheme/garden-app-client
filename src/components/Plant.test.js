import React from 'react';
import { shallow } from 'enzyme';
import Plant from './Plant';

describe('<Plant />', () => {

  it('should render page', () => {
    const fakePlant={
      _id: '123',
      plantName: 'Pepper'
    };
    const wrapper = shallow(<Plant plant={fakePlant}/>);
    expect(wrapper).toMatchSnapshot()
  })
})
