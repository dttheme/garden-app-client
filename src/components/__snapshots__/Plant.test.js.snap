// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Plant /> should render page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Plant
    deleteHandler={null}
    isNotLinking={false}
    numberPlanted={null}
    plant={
        Object {
            "_id": "123",
            "plantName": "Pepper",
          }
    }
    plantDate="00/00/00"
    plantLocation=""
    plantName=""
    waterFrequency=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="f4 fw9 tracked"
>
          <Link
                    className="washed-green no-underline underline-hover hover-light-yellow"
                    replace={false}
                    to="/plant/123"
          >
                    Pepper
          </Link>
           (
          )
</h3>,
        <p />,
        <p>
          Location: 
</p>,
        <p>
          You should water: 
</p>,
        null,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="washed-green no-underline underline-hover hover-light-yellow"
              replace={false}
              to="/plant/123"
>
              Pepper
</Link>,
            " (",
            undefined,
            ")",
          ],
          "className": "f4 fw9 tracked",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Pepper",
              "className": "washed-green no-underline underline-hover hover-light-yellow",
              "replace": false,
              "to": "/plant/123",
            },
            "ref": null,
            "rendered": "Pepper",
            "type": [Function],
          },
          " (",
          undefined,
          ")",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Location: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Location: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "You should water: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "You should water: ",
          undefined,
        ],
        "type": "p",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="f4 fw9 tracked"
>
            <Link
                        className="washed-green no-underline underline-hover hover-light-yellow"
                        replace={false}
                        to="/plant/123"
            >
                        Pepper
            </Link>
             (
            )
</h3>,
          <p />,
          <p>
            Location: 
</p>,
          <p>
            You should water: 
</p>,
          null,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="washed-green no-underline underline-hover hover-light-yellow"
                replace={false}
                to="/plant/123"
>
                Pepper
</Link>,
              " (",
              undefined,
              ")",
            ],
            "className": "f4 fw9 tracked",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Pepper",
                "className": "washed-green no-underline underline-hover hover-light-yellow",
                "replace": false,
                "to": "/plant/123",
              },
              "ref": null,
              "rendered": "Pepper",
              "type": [Function],
            },
            " (",
            undefined,
            ")",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Location: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Location: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "You should water: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "You should water: ",
            undefined,
          ],
          "type": "p",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
